cmd/collectors/rest/rest.go

```diff
@@ -306,7 +310,7 @@ func (r *Rest) processEndPoints() error {
 			i++
 		}
 
-		if records, err = r.GetRestData(endpoint.prop, r.client); err != nil {
+		if records, err = r.GetRestData(endpoint.prop.query, strings.Join(endpoint.prop.fields, ","), r.prop.returnTimeOut, r.client); err != nil {
```

Comment:
doesn't have to be fixed in this PR but this method call suggests there may be a better API something like this should work
```
r.Query(endpoint)
```
oh I read further in the commit and see this API was changed in this commit so I change my mind. This should probably be improved here :) at least for this case where an endpoint is being used